vue/wolox-books/src/views/Register.vue

```diff
@@ -0,0 +1,127 @@
+<template lang='pug'>
+  .register-container
+    img(alt='Vue logo' src='../assets/wolox_logo.svg' class='wolox-icon')
+    form(class='form-container' @submit.prevent="onSubmit")
+      p(class='title-form')
+        |{{labels.title}}
+      InputText(v-for='(field, index) in fields'
+        :name='field.name'
+        :label='field.label'
+        :key='index'
+        v-model='user[field.name]'
+      )
+      MainBtn(:label='labels.singUp')
+    MainBtn(:label='labels.singIn' class='login-button')
+</template>
+
+<script>
+import InputText from '@/components/InputText.vue'
+import MainBtn from '@/components/MainButton.vue'
+
+const fieldsArray = [
+  {
+    name: 'fisrtName',
+    label: 'First name'
+  },
+  {
+    name: 'lastName',
+    label: 'Last name'
+  },
+  {
+    name: 'email',
+    label: 'Email'
+  },
+  {
+    name: 'password',
+    label: 'Password'
+  }
+]
+
+const labels = {
+  title: 'BOOKS',
+  singUp: 'Sing up',
+  singIn: 'Login'
+}
+
+export default {
+  name: 'register',
+  props: {
+    fields: {
+      type: Array,
+      default: () => fieldsArray
+    },
+    labels: {
+      type: Object,
+      default: () => labels
+    }
+  },
+  components: {
+    InputText,
+    MainBtn
+  },
+  data () {
+    return {
+      user: {}
+    }
+  },
+  methods: {
+    onSubmit () {
+      console.log(this.user, 'user info')
+    }
+  }
+}
+</script>
+
+<style scoped lang='scss'>
+  @import 'src/scss/colors';
+  @import 'src/scss/fonts';
+
+  .register-container {
+    background-color: $wild-sand;
+    border-top: 5px solid $cerulean;
+    margin: 50px auto;
```

Comment:
use flex instead of auto

Reply:
Can you explain me why no use automatic margins?

Reply:
cause it's not the semantic usage of a margin, It's only to separate from its siblings. if this container will never have a sibling it's ok